src/core/chart-container.tsx

```diff
@@ -96,10 +96,11 @@ export function ChartContainer({
       ) : (
         <div
           style={chartMinWidth !== undefined ? { minInlineSize: chartMinWidth } : {}}
-          className={testClasses["chart-plot-wrapper"]}
+          className={clsx(styles["chart-plot-wrapper"], testClasses["chart-plot-wrapper"])}
         >
           {verticalAxisTitle}
           {chart(effectiveChartHeight)}
+          {!legend || legendPosition === "bottom" ? noData : null}
```

Comment:
This is another fix: previously, the no-data was centered around a larger block, so its position was affected by the filters in the top. Now it is no longer the case:

<img width="522" height="465" alt="Image" src="https://github.com/user-attachments/assets/2dda17a2-b55b-4369-891f-25be1c04d77b" />